synapse/federation/sender/per_destination_queue.py

```diff
@@ -516,6 +516,11 @@ async def _catch_up_transmission_loop(self) -> None:
                     # If the event is in the extremities, then great! We can just
                     # use that without having to do further checks.
                     room_catchup_pdus = [pdu]
+                elif self._store.is_partial_state_room(pdu.room_id):
+                    # We can't be sure which events the destination should
+                    # see using only partial state. Avoid doing so, and just retry
+                    # sending our the oldest PDU the remote is missing from us.
+                    room_catchup_pdus = [pdu]
```

Comment:
I think this makes sense after @reivilibre's explanation (modulo confusion about oldest vs newest).
The key bit I was missing before was that we're not filtering the catchup PDUs (which are okay to send out), but the forward extremities (which may not be okay to send out).